bugwarrior/services/__init__.py

```diff
@@ -481,22 +481,21 @@ def __repr__(self):
         return '<%s>' % self.__unicode__()
 
 
-class ServiceClient(object):
-    """ Abstract class responsible for making requests to service API's. """
-    @staticmethod
-    def json_response(response):
-        # If we didn't get good results, just bail.
-        if response.status_code != 200:
-            raise IOError(
-                "Non-200 status code %r; %r; %r" % (
-                    response.status_code, response.url, response.text,
-                ))
-        if callable(response.json):
-            # Newer python-requests
-            return response.json()
-        else:
-            # Older python-requests
-            return response.json
+def json_response(response):
+    """ Process a request's response to keep HTTP error handling consistent
+    accross services. """
+    # If we didn't get good results, just bail.
+    if response.status_code != 200:
+        raise IOError(
+            "Non-200 status code %r; %r; %r" % (
+                response.status_code, response.url, response.text,
+            ))
+    if callable(response.json):
+        # Newer python-requests
+        return response.json()
+    else:
+        # Older python-requests
+        return response.json
```

Comment:
I agree that it is "too much work to inherit from a mixin for a single static
method" but I disagree with this change.

I think that `ServiceClient` is a pattern starting to emerge in the services and that more opportunities to move logic out of the services and into `ServiceClient` will present themselves over time. If `ServiceClient` had been around for a while and no such opportunities had presented themselves, I would be more amenable to this change. But it hasn't been around very long and I think that the existence of this class will make those opportunities easier to identify and take advantage of.

Even without this DRY aspect, I think it facilitates the standardization of the service structure and makes it easier to make sense of the services that have a standalone `ServiceClient` -- ActiveCollab2, Github, Redmine, and TeamLab. It is not immediately obvious what the purpose of these client classes is, but you can look at their base class, read the docstring, and understand how their purpose is distinct from the other classes.